Source/HTTP/WinHttp/winhttp_provider.cpp

```diff
@@ -0,0 +1,556 @@
+#include "pch.h"
+#include "winhttp_provider.h"
+
+#if HC_PLATFORM == HC_PLATFORM_GDK
+#include <XGameRuntimeFeature.h>
+#endif
+#if HC_WINHTTP_WEBSOCKETS
+#include "Websocket/WinHTTP/winhttp_websocket.h"
+#endif
+
+NAMESPACE_XBOX_HTTP_CLIENT_BEGIN
+
+Result<std::shared_ptr<WinHttpProvider>> WinHttpProvider::Initialize()
+{
+    http_stl_allocator<WinHttpProvider> a{};
+    auto provider = std::shared_ptr<WinHttpProvider>{ new (a.allocate(1)) WinHttpProvider };
+
```

Comment:
isn't there some helper function like MakeShared that'd work here?

Reply:
That doesn't work with a private constructor.  I've marked the constructor private so that its clear that this class should be created/initialized with this method so that any initialization errors get reported.